src/main/java/com/aws/greengrass/shadowmanager/sync/model/LocalUpdateSyncRequest.java

```diff
@@ -121,24 +124,44 @@ public void execute(SyncContext context) throws SkipSyncRequestException, Confli
 
                 byte[] updatedDocument = response.getCurrentDocument();
                 long updateTime = Instant.now().getEpochSecond();
+                long localUpdatedVersion = getUpdatedVersion(response.getUpdateThingShadowResponse().getPayload())
+                        .orElse(currentLocalVersion + 1);
```

Comment:
What are the situations in which the `orElse` gets used?

Reply:
It is just a sanity check. Should be fine to have it.